### PR TITLE
OpenShift configuration issues

### PR DESCRIPTION
**Pull request by alesrosina**
As discussed here:
https://github.com/WeblateOrg/weblate/issues/2677#issuecomment-480742285

### CODE REVIEW

**Comment by alesrosina** [comment]
@nijel this is still WIP, so please do not merge it yet.

**Comment by alesrosina** [comment]
@GrahamDumpleton can you quickly review this, when you will have time?

**Comment by alesrosina** [comment]
@nijel i think this should actually cover the issue mentioned above - so from my side it is ready to merge. Maybe we can wait for a few more days, if @GrahamDumpleton will have time to quickly review it.

File: openshift3/persistent-postgresql.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
         {
             "name": "WEBLATE_VOLUME_SIZE",
             "description": "Size of the persistent volume for Weblate.",
-            "value": "1Gi",
+            "value": "40Gi",
```
Comment (nijel) [inline openshift3/persistent-postgresql.json]: Are these changes intentional? It heavily increases storage requirements...

Comment (alesrosina) [reply openshift3/persistent-postgresql.json]: Yes. `1gi` is OK for initial setup, but if you work on a project, that is bigger then 1GB (which is sadly today in `node` world quite often), this will obviously not work.

And 40GB today is really not a lot.

What do you think?

Comment (nijel) [reply openshift3/persistent-postgresql.json]: I have no clue what impact this has on OpenShift - force example if you can override that somehow or whether it just makes running Weblate unconditionally expensive. My guess is that most users won't need that much of space.

Comment (alesrosina) [reply openshift3/persistent-postgresql.json]: Fair enough, will lower it to 10Gi.

File: openshift3/persistent-mysql.json [hunk 2]
```diff
@@ -69,19 +69,19 @@
...
           "description": "Size of the persistent volume for Redis Cache.",
-          "value": "1Gi",
+          "value": "10Gi",
           "required": true
         },
         {
             "name": "DATABASE_VOLUME_SIZE",
             "description": "Size of the persistent volume for the database.",
-            "value": "1Gi",
+            "value": "40Gi",
```
Comment (alesrosina) [inline openshift3/persistent-mysql.json]: @nijel since we are talking about disk space - what is your experience, what would be average disk usage on database level?

File: openshift3/persistent-postgresql.json (hunk 1 above)
Comment (GrahamDumpleton) [reply openshift3/persistent-postgresql.json]: The whole point of it being a template parameter though is that people can override it when instantiating the template. In that sense, is better to set it to small value so test installs will work, with people who are doing production installs overriding it to what they believe they need. The alternative is to leave the field empty value, but with the field being required. This forces everyone to specify size as template parameter.

Setting allow expansion as default for PVC will also break for most people as not typical that OpenShift clusters would allow that. So deploy would presumably fail.

Key with the template is that it really should only serve as an example. For production use people should be copying it and customising it to their needs. So I would suggest leave it as 1Gi and not enable volume expansion.

Similar problem with memory. Now will not work for test installs.

My benchmark would be can it be installed in OpenShift Online starter tier.

Comment (GrahamDumpleton) [reply openshift3/persistent-postgresql.json]: The allow expansion could be a template parameter also with default of false.

File: openshift3/persistent-mysql.json (hunk 2 above)
Comment (nijel) [reply openshift3/persistent-mysql.json]: The whole https://hosted.weblate.org/ service consumes 8 GB in the database, that's why I think that 40 GB is way too much for default installation. Is this value easy to override when deploying? If yes, I'd stick with original small values, which would allow users to scale up, what is typically way easier than scaling down.

Comment (alesrosina) [reply openshift3/persistent-mysql.json]: OK, lowered it to 10Gi.

File: openshift3/persistent-postgresql.json (hunk 1 above)
Comment (alesrosina) [reply openshift3/persistent-postgresql.json]: @GrahamDumpleton OK, lowered it to 10Gi, i think this is reasonable. Also set `allowVolumeExpansion` to false.

Would that work?

**Comment by nijel** [comment]
Merged, thanks for your contribution!
